bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/model/LiveDataResponseTransformer.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solaredge.internal.model;
+
+import static org.openhab.binding.solaredge.internal.SolarEdgeBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.solaredge.internal.handler.ChannelProvider;
+import org.openhab.binding.solaredge.internal.model.LiveDataResponse.Connection;
+import org.openhab.binding.solaredge.internal.model.LiveDataResponse.SiteCurrentPowerFlow;
+import org.openhab.binding.solaredge.internal.model.LiveDataResponseMeterless.Overview;
+
+/**
+ * transforms the http response into the openhab datamodel (instances of State)
+ *
+ * @author Alexander Friese - initial contribution
+ */
+@NonNullByDefault
+public class LiveDataResponseTransformer extends AbstractDataResponseTransformer {
+    private static final Double ZERO_POWER = 0.0;
+
+    private final ChannelProvider channelProvider;
+
+    public LiveDataResponseTransformer(ChannelProvider channelProvider) {
+        this.channelProvider = channelProvider;
+    }
+
+    public Map<Channel, State> transform(LiveDataResponseMeterless response) {
+        Map<Channel, State> result = new HashMap<>(20);
+        Overview overview = response.getOverview();
+
+        if (overview != null) {
+            if (overview.currentPower != null) {
+                putPowerType(result, channelProvider.getChannel(CHANNEL_GROUP_LIVE, CHANNEL_ID_PRODUCTION),
+                        overview.currentPower.power, UNIT_W);
```

Comment:
These massive compiler warnings are not nice. You could store it to a local variable, which would be one extra line each.
```suggestion
            Power localCurrentPower = overview.currentPower;
            if (localCurrentPower != null) {
                putPowerType(result, channelProvider.getChannel(CHANNEL_GROUP_LIVE, CHANNEL_ID_PRODUCTION),
                        localCurrentPower.power, UNIT_W);
```

Reply:
Hmm it seems that I have overlooked this. In the other transformer class I solved it exactly like you suggested.